.github/workflows/static-analysis.yml

```
@@ -0,0 +1,103 @@
+  IWYU_BUILD: /home/iwyu/build
+  IWYU_SRC: /home/iwyu/include-what-you-use
+  IWYU_GIT_URL: https://github.com/include-what-you-use/include-what-you-use.git
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build:
+    runs-on: 'ubuntu-latest'
+
+    steps:
+      - name: Install python dependencies
+        run: |
+          python --version | grep -iE "python\s*3"  # fail if python 2 is default
+          pip    --version | grep -iE "python\s*3"
+          pip install cmake conan
+
+      - name: Install cppcheck
+        run: |
+          sudo apt-get update -qq
+          sudo apt-get install cppcheck
+
+      - name: Install default clang (clang-10 on Ubuntu 20.04)
+        if: env.USE_NEWER_CLANG == false
+        run: |
+          sudo apt get install clang llvm-dev libclang-dev clang-tidy
+
+      # TODO: use GH cache for clang and IWYU
+      - name: Install newer clang
+        if: env.USE_NEWER_CLANG == true
+        env:
+          LLVM_VER: 12
+        run: |
+          su && echo -e '\U0001F9B8' # 🦸
+          wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | apt-key add - 2>/dev/null
+          add-apt-repository -y "deb https://apt.llvm.org/bionic/ llvm-toolchain-bionic-${LLVM_VER} main"
+          apt-get update -qq
+          apt-get install -y --no-install-recommends \
+            clang-${LLVM_VER} lldb-${LLVM_VER} lld-${LLVM_VER} clangd-${LLVM_VER} \
+            llvm-${LLVM_VER}-dev libclang-${LLVM_VER}-dev clang-tidy-${LLVM_VER}
+          update-alternatives --install /usr/bin/clang-tidy clang-tidy $(which clang-tidy-${LLVM_VER}) 10
+          update-alternatives --install /usr/bin/clang   clang   $(which clang-${LLVM_VER}) 10
+          update-alternatives --install /usr/bin/clang++ clang++ $(which clang++-${LLVM_VER}) 10
+
+      - name: Install IWYU
+        # Alternately, https://github.com/EmilGedda/include-what-you-use-action is a thing
+        run: |
+          export LLVM_VER=$(clang --version | sed -n "s/^.*clang version \([0-9]\+\).*$/\1/p")
+          echo "LLVM_VER=${LLVM_VER}"
+          mkdir -p "${IWYU_BUILD}"
+          git clone --branch "clang_${LLVM_VER}" "${IWYU_GIT_URL}" "${IWYU_SRC}"
+          CC=clang-${LLVM_VER} CXX=clang++-${LLVM_VER} \
+            cmake -S ${IWYU_SRC} -B ${IWYU_BUILD} \
+            -G "Unix Makefiles" -DCMAKE_PREFIX_PATH=/usr/lib/llvm-${LLVM_VER}
+          cmake --build ${IWYU_BUILD} -j
+          cmake --install ${IWYU_BUILD}
+
+      - uses: actions/checkout@v2
+
+      - name: Cache
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-conan-modules
+        with:
+          path: |
+            ${{ env.CONAN_USER_HOME }}
+            ~/.cache/pip
+          key: ${{ runner.os }}-${{ env.BUILD_TYPE }}-${{ hashFiles('CMakeLists.txt') }}-${{ hashFiles('conanfile.py') }}
+
+      - name: Build and test
+        run: |
+          ctest --build-and-test . ./build_dir \
+                --build-generator "Unix Makefiles" -j2 \
```

A separate build step is better than relying on the unix specific make files.
```suggestion
                --build-generator "Unix Makefiles" -j2 \
```